From b373ca183ebe2c9088649e5e41905e586828d8cd Mon Sep 17 00:00:00 2001
From: Thales Elero Cervi <thaleselero.cervi@windriver.com>
Date: Thu, 8 Jun 2023 15:26:39 -0300
Subject: [PATCH] DPDK static build

The OVS-DPDK StarlingX OpenStack application use case requires that
openvswitch is statically linked against DPDK libraries.
This patch modifies the debian/rules accordingly.

StarlingX relies on dpdk-pmdinfo.py to verify which system NICs support
the current DPDK version, therefore the /usr/sbin/ovs-vswitchd ELF file
needs to be a static build so the ELF .rodata contains PMD vendor and
device information to be parsed.

---
 debian/rules | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/debian/rules b/debian/rules
index 3020b0315..92891df6e 100755
--- a/debian/rules
+++ b/debian/rules
@@ -32,7 +32,7 @@ ifneq (,$(filter i386 amd64 ppc64el arm64, $(DEB_HOST_ARCH)))
 		test -e Makefile || \
 		../configure --prefix=/usr --localstatedir=/var --enable-ssl --enable-shared \
 			--libdir=/usr/lib/$(DEB_HOST_MULTIARCH) \
-			--with-dpdk=shared --sysconfdir=/etc \
+			--with-dpdk=static --sysconfdir=/etc \
 			$(DATAPATH_CONFIGURE_OPTS))
 endif
 
-- 
2.25.1

